libr/anal/xrefs.c

```diff
@@ -43,31 +47,32 @@ R_API bool r_anal_xrefs_load(RAnal *anal, const char *prjfile) {
 		if (!db) return false;
 		path = strdup (db);
 	} else {
-		db = r_str_newf (R2_HOMEDIR"/projects/%s.d", prjfile);
+		db = r_str_newf ("%s/%s.d", prjdir, prjfile);
+		db_hack = db + 2;
 		if (!db) return false;
-		path = r_str_home (db);
+		path = r_str_home (db_hack);
 	}
 
 	if (!path) {
 		free (db);
 		return false;
 	}
 
-	ls_foreach (anal->sdb->ns, it, ns){
-		if (ns->sdb == DB){
-			ls_delete (anal->sdb->ns, it);
+	ls_foreach (core->anal->sdb->ns, it, ns){
+		if (ns->sdb == _DB){
+			ls_delete (core->anal->sdb->ns, it);
 			found = 1;
 			break;
 		}
 	}
-	if (!found) sdb_free (DB);
-	DB = sdb_new (path, "xrefs", 0);
-	if (!DB) {
+	if (!found) sdb_free (_DB);
+	_DB = sdb_new (path, "xrefs", 0);
+	if (!_DB) {
 		free (db);
 		free (path);
 		return false;
 	}
-	sdb_ns_set (anal->sdb, "xrefs", DB);
+	sdb_ns_set (core->anal->sdb, "xrefs", _DB);
```

Comment:
you cant use core from anal. this var is local to anal, so there's no need to reference it from core